openlibrary/templates/account/reading_log.html

```diff
@@ -24,14 +24,24 @@
 $add_metatag(property="og:description", content=og_description)
 $add_metatag(property="og:image", content=meta_photo_url)
 
+<form method="GET" class="olform pagesearchbox">
+  <input type="text" minlength="3" placeholder="Search by Title or Author" name="q" value="$(query_param('q', ''))"/>
+  <input type="submit"/>
+</form>
+
 <div class="mybooks-list">
+  $# If filtering results, override total_items to match the number of filtered results from bookshelves.get_users_logged_books().
+  $# This overrides pagination and will return all total_items on one page.
+  $if q:
+    $ total_items = len(items)
+        <span class="search-results-stats">$total_items hits</span>
   <span class="mybooks-tools"><img src="/images/icons/icon_sort.png" alt="$_('Sorting by')" style="margin-right:10px;" width="9" height="11">
     $if sort_order == 'desc':
       <strong class="lightgreen">$_("Date Added (newest)")</strong>
       |
-      <a href="$ctx.path?sort=asc">$_("Date Added (oldest)")</a>
+      <a href="$ctx.path?sort=asc&q=$q">$_("Date Added (oldest)")</a>
```

Comment:
I believe this keeps all the other params and just changes the one in question, but we should test

```suggestion
      <a href="$changequery(sort='asc')">$_("Date Added (oldest)")</a>
```

Reply:
I tested this and it seems to work. It seems I need to read more about the web.py API!